src/lib/registration/ui/migration_repos_selection_dialog.rb

```diff
@@ -0,0 +1,140 @@
+# ------------------------------------------------------------------------------
+# Copyright (c) 2015 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# ------------------------------------------------------------------------------
+#
+
+require "yast"
+
+module Registration
+  module UI
+    # dialog for manual migration repository selection
+    class MigrationReposSelectionDialog
+      include Yast::Logger
+      include Yast::I18n
+      include Yast::UIShortcuts
+
+      Yast.import "Pkg"
+      Yast.import "Wizard"
+
+      attr_accessor :selected_migration, :manual_repo_selection
+
+      # display and run the dialog
+      # @return [Symbol] user input
+      def self.run
+        dialog = MigrationReposSelectionDialog.new
+        dialog.run
+      end
+
+      # constructor
+      def initialize
+        textdomain "registration"
+      end
+
+      # display and run the dialog
+      # @return [Symbol] user input
+      def run
+        Yast::Wizard.SetContents(
+          # TRANSLATORS: dialog title
+          _("Migration Repositories"),
+          dialog_content,
+          # TRANSLATORS: help text
+          # FIXME: help text
+          _("FIXME"),
```

Comment:
I think we should have at least something useful here before merge otherwise we can end with it in final product :)